### PR TITLE
Species Component

### PR DESCRIPTION
**Pull request by clusterfack**
Wew what an ongoing shitshow this feature has been. I wrote this in such a shitty way before and had to scrap a good portion of it to make it less awful. 

This is part 1 of x in the ongoing mob component series. This is meant to control the major life states of mobs which just about each and every mob should have as a fundamental component which is why its so limited in scope. This component is meant to be a distinct component from the following planned mob features/components:
- Status effects (stun, knockdown, etc)
- Organs (internal, external)
- ReagentMetabolizer
- Environment (pressure/temperature)
- Food/Stomach
- Surgical

Things that should be done before merging this:
- ~~Fix the hud change message not being received on the client for what the fuck ever reason (I cant figure it out)~~
- ~~Merge this species UI with the inventory UI somehow into a single window~~
- ~~Add a screen shader that gets applied on crit/death~~
- ~~Make screen shader not cover UI~~

### CODE REVIEW

File: Content.Server/GameObjects/Components/Damage/DamageableComponent.cs [hunk 1]
```diff
@@ -18,26 +18,27 @@ namespace Content.Server.GameObjects
...
         /// <summary>
         /// The resistance set of this object.
         /// Affects receiving damage of various types.
         /// </summary>
         [ViewVariables]
         public ResistanceSet Resistances { get; private set; }
 
-        Dictionary<DamageType, int> CurrentDamage = new Dictionary<DamageType, int>();
-        Dictionary<DamageType, List<int>> Thresholds = new Dictionary<DamageType, List<int>>();
+        public Dictionary<DamageType, int> CurrentDamage = new Dictionary<DamageType, int>();
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Damage/DamageableComponent.cs]: Not sure that it's a good idea to make this field public. Wouldn't some proper encapsulation be better?

File: Content.Server/GameObjects/Components/Damage/ResistanceSet.cs [hunk 2]
```diff
@@ -7,16 +8,16 @@ namespace Content.Server.GameObjects
...
-        Total,
-        Brute,
-        Heat,
-        Cold,
-        Acid,
-        Toxic,
-        Electric
-    }
+    //public enum DamageType
+    //{
```
Comment (PJB3005) [inline Content.Server/GameObjects/Components/Damage/ResistanceSet.cs]: Just remove this if you're gonna move it.

**Comment by clusterfack** [comment]
Current WIP image, nearly completed except for conflict resolving and the screen shader issue

**Comment by clusterfack** [comment]
Cool finished

**Pull request merged**